### DISCUSSION TITLE
Customizing mini.pairs for Rust

### DISCUSSION THREAD

**Discussion by ghost** [LazyVim/LazyVim#1931] [Q&A] [answered]
Wondering if anyone has overwritten the default configurations for single quote pairs `'` to not create a pair when defining lifetimes. Might just be easier to disable the single quote pairing in Rust files altogether. I'm not sure sure how to do that either though.

**Comment by dpetka2001** [comment, marked as answer]
I tried the following autocmd in `/lua/config/autocmds.lua`
```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = "lua",
  group = vim.api.nvim_create_augroup("Rust_disable_single_quote", { clear = true }),
  callback = function()
    MiniPairs.unmap("i", "'", "''")
  end,
  desc = "Disable single quote Rust",
})
```
I tested it with Lua because I don't do Rust. But if you change the pattert to the Rust filetype it should work for Rust I believe.

**Reply by ghost** [reply]
Wow, thank you for sharing your knowledge. That works as expected, I appreciate you.

**Comment by dydyamotya** [comment]
Hi! 

I've noticed that this recipe:

> I tried the following autocmd in `/lua/config/autocmds.lua`
> 
> ```lua
> vim.api.nvim_create_autocmd("FileType", {
>   pattern = "lua",
>   group = vim.api.nvim_create_augroup("Rust_disable_single_quote", { clear = true }),
>   callback = function()
>     MiniPairs.unmap("i", "'", "''")
>   end,
>   desc = "Disable single quote Rust",
> })
> ```
> 
> I tested it with Lua because I don't do Rust. But if you change the pattert to the Rust filetype it should work for Rust I believe.

doesn't work for the first opened rust file, because of lazy loading.

I've added autocmd for handling such a situation:

```
local rust_group = vim.api.nvim_create_augroup("Rust_disable_single_quote", { clear = true })

vim.api.nvim_create_autocmd("FileType", {
  pattern = "rust",
  group = rust_group,
  callback = function()
    if MiniPairs ~= nil then
      MiniPairs.unmap("i", "'", "''")
    end
  end,
  desc = "Disable single quote Rust",
})

vim.api.nvim_create_autocmd("User", {
  pattern = "LazyLoad",
  group = vim.api.nvim_create_augroup("Rust_disable_single_quote_lazyload", { clear = true }),
  callback = function(event)
    if event.data == "mini.pairs" then
      vim.api.nvim_exec_autocmds("FileType", {
        pattern = "rust",
        group = rust_group,
      })
    end
  end,
  desc = "Disable single quote Rust",
})
```

**Comment by 4ydx** [comment]
Regarding disabling entirely, just create/append to a disabled.lua file.  You can of course call the file whatever you like.

```
cat .config/nvim/lua/plugins/disabled.lua
return {
  { "echasnovski/mini.pairs",  enabled = false },
}
```

**Comment by kahojyun** [comment]
I found a solution provided by the author of `mini.pairs` on [reddit](https://www.reddit.com/r/neovim/comments/171xezk/disable_minipair_based_on_buffer_file_type/). This command will only disable single-quote pairs for Rust buffers; other non-Rust buffers won't be affected.

```lua
-- lua/config/autocmds.lua
local function augroup(name)
  return vim.api.nvim_create_augroup("custom_" .. name, { clear = true })
end

vim.api.nvim_create_autocmd({ "FileType" }, {
  group = augroup("rust_disable_single_quote_pairs"),
  pattern = "rust",
  callback = function()
    vim.keymap.set("i", "'", "'", { buffer = 0 })
  end,
})
```
